### PR TITLE
Add "docfx --version", adjust command line help text

### PR DESCRIPTION
**Pull request by yufeih**

### CODE REVIEW

File: src/docfx/cli/Program.cs [hunk 1]
```diff
@@ -67,20 +73,25 @@ private static (string command, string docset, CommandLineOptions options) Parse
...
+                syntax.DefineParameter("docset", ref docset, "Docset directory that contains docfx.yml.");
 
                 // Build command
                 // usage: docfx build [docset] [-o/--out output] [-l/--log log] [--legacy]
-                syntax.DefineCommand("build", ref command, "builds a folder containing docfx.yml");
-                syntax.DefineOption("o|out", ref options.Output, "output folder");
-                syntax.DefineOption("l|log", ref options.Log, "path to log file");
-                syntax.DefineOption("legacy", ref options.OutputLegacyModel, "output legacy model for backward compatibility");
-                syntax.DefineParameter("docset", ref docset, "docset path that contains docfx.yml");
+                syntax.DefineCommand("build", ref command, "Builds a docset.");
```
Comment (superyyrrzz) [inline src/docfx/cli/Program.cs]: why remove log?

Comment (yufeih) [reply src/docfx/cli/Program.cs]: fixed

File: src/docfx/cli/Program.cs [hunk 2]
```diff
@@ -67,20 +73,26 @@ private static (string command, string docset, CommandLineOptions options) Parse
...
-                syntax.DefineCommand("restore", ref command, "restores dependencies before build");
-                syntax.DefineParameter("docset", ref docset, "docset path that contains docfx.yml");
+                syntax.DefineCommand("restore", ref command, "Restores dependencies before build.");
+                syntax.DefineParameter("docset", ref docset, "Docset directory that contains docfx.yml.");
 
                 // Build command
-                // usage: docfx build [docset] [-o/--out output] [-l/--log log] [--legacy]
-                syntax.DefineCommand("build", ref command, "builds a folder containing docfx.yml");
-                syntax.DefineOption("o|out", ref options.Output, "output folder");
-                syntax.DefineOption("l|log", ref options.Log, "path to log file");
```
Comment (superyyrrzz) [inline src/docfx/cli/Program.cs]: `l|log`?

Comment (yufeih) [reply src/docfx/cli/Program.cs]: `log` is short enough, I'd like to reserve `l` for something else

**Pull request merged**